providers/aws/resources/aws_iam.go

```diff
@@ -852,10 +853,10 @@ func (a *mqlAwsIamUser) attachedPolicies() ([]interface{}, error) {
 
 		for i := range userAttachedPolicies.AttachedPolicies {
 			attachedPolicy := userAttachedPolicies.AttachedPolicies[i]
-
 			mqlAwsIamPolicy, err := CreateResource(a.MqlRuntime, "aws.iam.policy",
-				map[string]*llx.RawData{"arn": llx.StringDataPtr(attachedPolicy.PolicyArn)},
-			)
+				// this isn't correct for the id, this is the name. we need to remove "id" from the policy type.
+				// it is creating a conflict bc we cannot make it optional, as it then bumps up against the internal id
+				map[string]*llx.RawData{"arn": llx.StringDataPtr(attachedPolicy.PolicyArn), "id": llx.StringDataPtr(attachedPolicy.PolicyName)})
```

Comment:
is anyone has ideas on how to better handle this, I'm all ears!

Reply:
Just an idea, I'm not sure what happens:
What if you set the "id" field to `PolicyArn`. Then it would at least be the same data as what is returned by `id()`.